keanu-project/src/test/java/io/improbable/keanu/util/csv/WriteCsvTest.java

```diff
@@ -0,0 +1,248 @@
+package io.improbable.keanu.util.csv;
+
+import io.improbable.keanu.algorithms.NetworkSamples;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.intgr.nonprobabilistic.ConstantIntegerVertex;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.*;
+
+import static org.junit.Assert.assertTrue;
+
+public class WriteCsvTest {
+
+    private NetworkSamples samples;
+    private List<Vertex<DoubleTensor>> rowTensors = new ArrayList<>();
+    private List<Vertex<DoubleTensor>> columnTensors = new ArrayList<>();
+    private List<Vertex<DoubleTensor>> scalarTensors = new ArrayList<>();
+    private List<Vertex<IntegerTensor>> integerColumnTensors = new ArrayList<>();
+
+    @Before
+    public void setup() {
+        GaussianVertex g1 = new GaussianVertex(new int[]{1, 5}, 0, 1);
+        GaussianVertex g2 = new GaussianVertex(new int[]{1, 4}, 0, 1);
+
+        GaussianVertex f1 = new GaussianVertex(new int[]{5, 1}, 0, 1);
+        GaussianVertex f2 = new GaussianVertex(new int[]{4, 1}, 0, 1);
+
+        g1.setValue(new double[]{1, 2, 3, 4, 5});
+        g2.setValue(new double[]{5, 4, 3, 2});
+
+        f1.setValue(new double[]{1, 2, 3, 4, 5});
+        f2.setValue(new double[]{5, 4, 3, 2});
+
+        ConstantDoubleVertex c1 = new ConstantDoubleVertex(0.5);
+        ConstantDoubleVertex c2 = new ConstantDoubleVertex(1.5);
+
+        ConstantIntegerVertex i1 = new ConstantIntegerVertex(new int[]{1, 2, 3});
+        ConstantIntegerVertex i2 = new ConstantIntegerVertex(new int[]{3, 2, 1});
+
+        rowTensors.addAll(Arrays.asList(g1, g2));
+        columnTensors.addAll(Arrays.asList(f1, f2));
+        scalarTensors.addAll(Arrays.asList(c1, c2));
+        integerColumnTensors.addAll(Arrays.asList(i1, i2));
+
+        Map<Long, List<DoubleTensor>> networkSamples = new HashMap<>();
+        networkSamples.put(g1.getId(), Arrays.asList(g1.getValue(), g1.times(2).getValue()));
+        networkSamples.put(g2.getId(), Arrays.asList(g2.getValue(), g2.times(2).getValue()));
+        samples = new NetworkSamples(networkSamples, 2);
+    }
+
+    @Test
+    public void writeSamplesToCsv() {
+        File file = WriteCsv.asSamples(samples, rowTensors).toFile("test.csv");
```

Comment:
not sure where this file will go - might be worth using a temporary file: https://docs.oracle.com/javase/7/docs/api/java/io/File.html#createTempFile%28java.lang.String,%20java.lang.String%29
which will get automatically cleaned up even in the event of the code crashing.

Reply:
nice-  changed to be a temp file that's created